### ISSUE TITLE
Hazard.set_raster with single .tif file

### ISSUE THREAD

**Issue by gregort42** [CLIMADA-project/climada_python#251]
```
test_file1 = Path("G:\Testfile1_2\Testfile1.tif")
haz = Hazard('AV')
haz.set_raster(test_file1)
```

Get the test files here: https://we.tl/t-562rNSzvNi

FYI testfile1 has diferent extents than testfile2, but reading only testfile1 would be enogh for the moment.

**Comment by chahank** [comment]
Could you please explain what is the issue? What is testfile2?

**Comment by tovogt** [comment]
@gregort42: A `Hazard` object contains the impacts from several events (instead of only one). That's why the `set_raster` method expects a list of files instead of only a single file. Please use `haz.set_raster([test_file1])`.

**Comment by gregort42** [comment]
So testfile1 is a simulated avalalche file of one region (and extent), testfile2 is a file of another region (and with another extent) just for the case we want to give it a trial to solve the "more regions" issiue

After creating the hazard  ("AV")  with single files as well as with a list of files get this error: 
there must be somthing wrong with the dimension
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-139-def48d5d0673> in <module>
----> 1 haz = create_AV_hazard_tif(300, 'Buchs', crs='WGS84')

<ipython-input-133-161495f6923b> in create_AV_hazard_tif(rp, region, year, crs)
      3     file_names = get_file_names(str(folder / (region + '*.tif'))) # get all file_names in a list
      4     haz = Hazard('AV')
----> 5     haz.set_raster(file_names) # initialize the hazard (centroids etc) with the first file
      6     haz.event_id = np.array([int(x.split('_')[-2]) for x in file_names])
      7     haz.frequency = np.repeat(1.0/rp,haz.size)# defining the return period

~\Desktop\climada_python-main210517\climada\hazard\base.py in set_raster(self, files_intensity, files_fraction, attrs, band, src_crs, window, geometry, dst_crs, transform, width, height, resampling)
    260                     file, band, src_crs, window, geometry, dst_crs, transform,
    261                     width, height, resampling))
--> 262             self.intensity = sparse.vstack(inten_list, format='csr')
    263             if files_fraction is not None:
    264                 fract_list = []

~\AppData\Local\Continuum\anaconda3\envs\climada_env210\lib\site-packages\scipy\sparse\construct.py in vstack(blocks, format, dtype)
    499 
    500     """
--> 501     return bmat([[b] for b in blocks], format=format, dtype=dtype)
    502 
    503 

~\AppData\Local\Continuum\anaconda3\envs\climada_env210\lib\site-packages\scipy\sparse\construct.py in bmat(blocks, format, dtype)
    548 
    549     if blocks.ndim != 2:
--> 550         raise ValueError('blocks must be 2-D')
    551 
    552     M,N = blocks.shape

ValueError: blocks must be 2-D

**Comment by chahank** [comment]
@gregort42 : have you tried the solution proposed by @tovogt ?

@tovogt : if this is the solution, could we add a useful Error message or simply replace the single file by a list by default?

**Comment by tovogt** [comment]
@chahank: I already pushed a fix: b98fc8e

@gregort42: Your original example code works now. For your new example, you need to provide more information about your setup. Would you please create a minimal example with tif-files so that we are able to reproduce your error?

**Comment by chahank** [comment]
Amazing @tovogt !

**Comment by gregort42** [comment]
> 
> 
> @chahank: I already pushed a fix: [b98fc8e](https://github.com/CLIMADA-project/climada_python/commit/b98fc8e5a4e7446c407fd9e6513b860dd52379d7)
> 
> @gregort42: Your original example code works now. For your new example, you need to provide more information about your setup. Would you please create a minimal example with tif-files so that we are able to reproduce your error?

@tovogt: Thomas thanks a lot for your effort your fix solved my issues !
